explainer.md

```diff
@@ -102,6 +102,10 @@ When the user uninstalls this web application, registered file handlers and icon
 
 For more advanced use cases, such as opening a file in an existing window or displaying a notification, applications can add a [launch event handler](https://github.com/WICG/sw-launch/blob/master/explainer.md).
 
+## Permissions API Integration
+
+A `"file-handling"` permission will be defined, in order to protect the user from accidentally allowing a PWA to view files. This permission prompt may asynchronously show right after the user chooses to use a PWA capabile of handling the associated file type, to open a file, so that the permission context is more understandable and relevant. This permission prompt should also show after the PWA has an opportunity to load, and the prompt should clearly show the PWA origin, so that it is clear to the user the identity of the PWA requesting file read access for the handled file. Launched files should only be queued into the `launchQueue` after the user accepts the permission and allows the PWA to view the relevant file types. User agents may retain control over default settings and how (or if) they are exposed to the user.
```

Comment:
```suggestion
A `"file-handling"` permission will be defined to protect the user from accidentally allowing a PWA to view files. This permission prompt may asynchronously show right after the user chooses to use a PWA to open a file of any of the file types the PWA tries to associate with, so that the permission context is more understandable and relevant. This permission prompt should also show after the PWA has had an opportunity to load, and the prompt should clearly show the PWA's origin, so that the identity of the PWA requesting file read access for the handled file is clear to the user. Launched files should only be queued into the `launchQueue` after the user accepts the permission and allows the PWA to handle the relevant file types. User agents may retain control over default settings and how (or if) they are exposed to the user.
```